c_src/sasl_auth.c

```diff
@@ -481,7 +507,7 @@ static ERL_NIF_TERM sasl_srv_new(ErlNifEnv* env, int UNUSED(argc), const ERL_NIF
     memcpy(state->callbacks, callbacks, sizeof(callbacks));
 
     int result = sasl_server_new(
-        (const char*)state->service, NULL, NULL, NULL, NULL, state->callbacks, 0, &state->conn);
+        (const char*)state->service, (const char*)state->host, NULL, NULL, NULL, state->callbacks, 0, &state->conn);
```

Comment:
if result is not SASL_OK, should the state get destroyed?

Reply:
yes, and it's released at line 524?

Reply:
The `state` won't get destroyed because we `make_resource` in L511 which leaves a lingering `term` that won't get returned to the erlang for future GC and that hold one refcnt. 

I think `erl_make_resource` should only be called for `SASL_OK`